src/interval-tree.h

```diff
@@ -0,0 +1,789 @@
+/*	$NetBSD: interval-tree.h,v 1.8 2004/03/28 19:38:30 provos Exp $	*/
```

Comment:
I think it would be good to add some info here:
- license of the file
- where did you take it from? Seems to come from one of the BSD's?

Reply:
Please note that:
- There is the 2-clause BSD license a bit down further.
- I copied the pre-existing `tree.h` from `master` in this commit.
- I modified it for interval tree ops in the later commits.
- Currently, the `tree.h` that exists in *BSD is very different from this one. They implement rank balanced trees instead of red black trees (which is currently used by our code and these interval trees). So, idk if it's right to say that it comes from BSDs.

So, questions for you:
1. What should I add as license if the license is written down below?
2. Should I add that the file came from older versions of BSD`s `tree.h` to both `tree.h` and `interval-tree.h`?

Reply:
ah no you can mention it's a copy of the in-tree tree.h